launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/docker/launcher/ContainerCommandLauncher.java

```diff
@@ -308,6 +308,12 @@ public Process execute(IPath commandPath, String[] args, String[] env, IPath wor
 		additionalDirs.addAll(additionalPaths.stream().map(p -> ContainerLaunchUtils.toDockerVolume(pathMap, p))
 				.collect(Collectors.toList()));
 
+		final var getImg = ContainerLaunchUtils.provideDockerImage(monitor, connectionName, imageName);
+		if (getImg != null) {
+			setErrorMessage(getImg);
+			return null;
+		}
+
```

Comment:
How do you sync between the time it takes to pull the image and running the command below?  For example, a very large image may take a long time and won't be available by the time the subsequent command gets called.

Reply:
provideDockerImage is blocking. That's why `monitor` is passed.

Reply:
Ok.